### PR TITLE
Autoscoped constructors in a vector

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Tests for autoscoped constructors in a vector and allowing comments in multiline expression block.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      style guides.blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

File: test/Table_Tests/src/Helpers/Auto_Constructor_Instrumentation_Spec.enso [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                enter  788f7b33-54ae-4452-85ff-4b7b5688b598
+                result 788f7b33-54ae-4452-85ff-4b7b5688b598 result: Aggregate_Column type: Aggregate_Column.type
+                enter  d1844837-30e7-46b7-bde3-72f7afec52cf
+                result d1844837-30e7-46b7-bde3-72f7afec52cf result: UnresolvedSymbol<Count> type: Function
+                enter  00e33517-6d19-4f5a-84c5-126f59a93b67
+                callfn 00e33517-6d19-4f5a-84c5-126f59a93b67 fn: null self=_ args: [Aggregate_Column]
+                result 00e33517-6d19-4f5a-84c5-126f59a93b67 result: (Count '') type: Aggregate_Column
+
+            pair.first.column_names . should_equal pair.second.column_names
+            pair.first.column_names.map n->
```
Comment (JaroslavTulach) [inline test/Table_Tests/src/Helpers/Auto_Constructor_Instrumentation_Spec.enso:32]: The new test verifies that both `Table`s contain the same elements - e.g. resolution of _autoscoped constructors_ works fine in runtime thanks to 5f9aa2510dd9f73c3009396d59b1a426acd0ed10 change.

File: test/Table_Tests/src/Helpers/Table_Aggregate_Auto_Constructor.enso [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+perform print=IO.println =
+    row1 = [ 3 ]
+    row2 = [ 2 ]
+    row3 = [ 3 ]
+    operator16985 = Table.from_rows [ 'Organisation' ] [row1, row2, row3 ]
+
+    interesting_uuids = Vector.new_builder
+        # line 9: (..Count)
+        . append "4cf46843-236f-4551-a9cc-30dd7552bb5f" # 287, 1
```
Comment (JaroslavTulach) [inline test/Table_Tests/src/Helpers/Table_Aggregate_Auto_Constructor.enso:21]: Comments in the 
- #8415

resulted in `NullPointerException` and required 9cc208dc2e1cd235ce24bc499bb8f2c235efea76 fix.

File: distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Helper.enso [hunk 3]
```diff
@@ -153,8 +153,7 @@ all_same_column aggregates =
 
    If some columns cannot be resolved, `Nothing` is returned. Any issues are
    reported to the `Problem_Builder`.
-resolve_aggregate : Table -> Problem_Builder -> Aggregate_Column -> Aggregate_Column | Nothing
-resolve_aggregate table problem_builder aggregate_column =
+resolve_aggregate table:Table problem_builder:Problem_Builder aggregate_column:Aggregate_Column =
```
Comment (jdunkerley) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Helper.enso]:
```suggestion
resolve_aggregate table:Table problem_builder:Problem_Builder aggregate_column:Aggregate_Column -> Aggregate_Column | Nothing =
```

Comment (jdunkerley) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Helper.enso]: As we are removing the other line we should add the return type as well.

**Review by jdunkerley — APPROVED** [review]
one slight change pls

**Review by hubertp — APPROVED** [review]

File: distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Helper.enso (hunk 3 above)
Comment (JaroslavTulach) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Helper.enso]: There are [test failures](https://github.com/enso-org/enso/actions/runs/8553039311/job/23435393626?pr=9630#step:8:8194) caused by too strict typing (for example `table` isn't `Table` from time to time).

Trying different approach: 5f9aa2510d

Comment (jdunkerley) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Helper.enso]: Yes `table` can be a `DB_Table` but we can't end up with a circle here.

We should create a conversion type `Table` and `DB_Table` can merge into but we haven't yet.

Comment (JaroslavTulach) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Helper.enso]: I leave that for another libs PR then. I am fine with 5f9aa2510dd9f73c3009396d59b1a426acd0ed10 - everything is green.

**Pull request merged**